test/test-raja-perf-suite.cpp

```diff
@@ -0,0 +1,120 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-22, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "gtest/gtest.h"
+
+#include "common/Executor.hpp"
+#include "common/KernelBase.hpp"
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <cmath>
+
+TEST(ShortSuiteTest, Basic)
+{
+
+// Assemble command line args for basic test
+  int argc = 4;
+#if defined(RAJA_ENABLE_HIP)
+  argc = 6;
+#endif
+  std::vector< std::string > sargv(argc);
+  sargv[0] = std::string("dummy ");  // for executable name
+  sargv[1] = std::string("--checkrun ");
+  sargv[2] = std::string("5 ");
+  sargv[3] = std::string("-show-progress ");
+#if defined(RAJA_ENABLE_HIP)
+  sargv[4] = std::string("--exclude-kernels ");
+  sargv[5] = std::string("HALOEXCHANGE_FUSED");
+#endif
+
+  char** argv = new char* [argc];
+  for (int is = 0; is < argc; ++is) { 
+    argv[is] = const_cast<char*>(sargv[is].c_str());
```

Comment:
Would it make more sense to make executor take const char** instead? We'd have to add a cast in main, but it could avoid const_cast here.

Reply:
Or have two constructors for Executor to hide the cast of char** to const char** there.

Reply:
I'm not sure it makes much difference. Again, following the common practice....

`int main( int argc, char** argv )`

Reply:
Ya, const_cast is just evil.

Reply:
I was trying to make the code simpler using C++ strings. c_str() returns const char*

Reply:
Its fine, I was just hoping for an easy way to avoid the const_cast.

Reply:
i kinda like it. it makes me feel naughty.